mlx/backend/metal/device.cpp

```diff
@@ -357,7 +357,6 @@ MTL::Function* Device::get_function_(
   }
 
   mtl_func_consts->release();
```

Comment:
@jagrit06 I was getting segfaults caused by this release when using function constants, but couldn't figure out the best place in the code to move it to. Any idea where it should fit in?

Reply:
I think it's a bug to release that, and deleting it is correct. https://github.com/bkaradzic/metal-cpp/blob/metal-cpp_macOS14.2_iOS17.2/README.md#memory-allocation-policy